packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2407,7 +2407,14 @@ class RenderOffstage extends RenderProxyBox {
     if (value == _offstage)
       return;
     _offstage = value;
-    markNeedsLayout();
+    markNeedsLayout(changedSizedByParent: true);
+  }
+
+  @override
+  void markNeedsLayout({ bool changedSizedByParent: false }) {
```

Comment:
It would be more efficient if you didn't do this as an override.  The default value is just to call through to the super class, so you've made this polymorphic with needing to.  Instead, `set offstage` can call `markParentNeedsLayout` directly.